distributed/joblib.py

```diff
@@ -35,5 +35,11 @@ def callback_wrapper():
         future.get = future.result  # monkey patch to achieve AsyncResult API
         return future
 
+    def abort_everything(self, ensure_ready=True):
+        # Tell the executor to cancel any task submitted via this instance
+        # as joblib.Parallel will never access those results.
+        executor = self.executor
+        executor.cancel([Future(k, executor) for k in executor.futures])
```

Comment:
This would also allow you to test the early abort functionality by asserting that the status of all futures was `cancelled`.